src/components/TupleSelect/index.tsx

```diff
@@ -61,31 +65,80 @@ export const TupleSelect: React.VFC<TupleSelectProps> = ({
               {o.displayName}
             </Option>
           ))}
-        </Select>
-        <Input
+        </GroupedSelect>
+        <GroupedTextInput
+          aria-labelledby="filter-input"
           id="filter-input"
-          aria-label="Select Text Input"
           data-cy="tuple-select-input"
           value={input}
-          onChange={(e) => setInput(e.target.value)}
-          style={{ width: "70%" }}
+          onChange={(e) => handleOnChange(e.target.value)}
           placeholder={selectedOption.placeHolderText}
-          suffix={
-            <Icon
-              glyph="Plus"
-              onClick={handleOnSubmit}
-              aria-label="Select plus button"
-              data-cy="tuple-select-button"
-            />
+          onKeyPress={(e: React.KeyboardEvent<HTMLInputElement>) =>
+            e.key === "Enter" && handleOnSubmit()
+          }
+          icon={
+            <div>
+              {isValid ? (
+                <IconButton
+                  onClick={handleOnSubmit}
+                  aria-label="Select plus button"
+                >
+                  <Icon glyph="Plus" data-cy="tuple-select-button" />
+                </IconButton>
+              ) : (
+                <InactiveIconWrapper>
+                  <IconTooltip
+                    glyph="Warning"
+                    data-cy="tuple-select-warning"
+                    fill={yellow.base}
+                  >
+                    {validatorErrorMessage}
+                  </IconTooltip>
+                </InactiveIconWrapper>
+              )}
+            </div>
           }
-          onPressEnter={handleOnSubmit}
         />
-      </Input.Group>
+      </InputGroup>
     </Container>
   );
 };
 
 const Container = styled.div`
   display: flex;
   flex-direction: column;
+  position: relative;
+`;
+
+const InputGroup = styled.div`
+  align-items: center;
+  display: flex;
+  flex-direction: row;
 `;
+
+const InactiveIconWrapper = styled.div`
+  margin: 6px;
+`;
```

Comment:
This seems kind of hacky to me... 🤔

A less hacky solution (but still not that pretty) would be to put a fixed `width` & `justify-content: center` property in the `IconWrapper` style in `TextInputWithGlyph`

<img width="464" alt="Screen Shot 2022-04-26 at 9 23 29 PM" src="https://user-images.githubusercontent.com/47064971/165419570-36026e6e-376b-4895-8605-8991f0d23b19.png">

result:

https://user-images.githubusercontent.com/47064971/165419665-d696856c-4e4c-434e-9521-95001387f285.mov